h2integrate/core/h2integrate_model.py

```diff
@@ -455,18 +456,48 @@ def create_finance_model(self):
                         "Available "
                         f"technologies: {list(self.technology_config['technologies'].keys())}"
                     )
+            if commodity_stream is not None:
+                if "combiner" not in commodity_stream and commodity_stream not in tech_names:
+                    raise UserWarning(
+                        f"The technology specific for the commodity_stream '{commodity_stream}' "
+                        f"is not included in subgroup '{subgroup_name}' technologies list."
+                        f" Subgroup '{subgroup_name}' includes technologies: {tech_names}."
+                    )
 
             finance_subgroups.update(
                 {
                     subgroup_name: {
                         "tech_configs": tech_configs,
                         "commodity": commodity,
+                        "commodity_stream": commodity_stream,
                     }
                 }
             )
             finance_subgroup = om.Group()
 
-            if commodity == "electricity":
+            if commodity_stream is not None:
+                commodity_summer_model = self.supported_models.get("production_summer")  # FINISH
+                if "combiner" in commodity_stream or "splitter" in commodity_stream:
+                    commodity_summer_config = self.technology_config["technologies"][
+                        commodity_stream
+                    ]
+                else:
+                    commodity_summer_config = {
+                        "model_inputs": {
+                            "performance_parameters": {
+                                "commodity": commodity,
+                                "commodity_units": "kW" if commodity == "electricity" else "kg",
+                            }
+                        }
+                    }
+                commodity_summer = commodity_summer_model(
+                    driver_config=self.driver_config,
+                    plant_config=self.plant_config,
+                    tech_config=commodity_summer_config,
+                )
+                finance_subgroup.add_subsystem(f"{commodity}_sum", commodity_summer)
```

Comment:
Could you please add comments throughout this section about what's going on here?